bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/SyncTableValidationOnlyJob.java

```diff
@@ -0,0 +1,1102 @@
+            + " --targetbigtableinstance=instance123 --sourcezkcluster=zk1.example.com,zk2.example.com,zk3.example.com:2181:/hbase"
+            + " gs://bucket/hashes/tableA tableA tableA");
+  }
+
+  @Override
+  public int run(String[] args) throws Exception {
+    String[] otherArgs = new GenericOptionsParser(getConf(), args).getRemainingArgs();
+    if (!doCommandLine(otherArgs)) {
+      return 1;
+    }
+
+    Job job = createSubmittableJob(otherArgs);
+    if (!job.waitForCompletion(true)) {
+      return 1;
+    }
+    counters = job.getCounters();
+    return 0;
+  }
+
+  protected boolean doCommandLine(String[] args) {
+    if (!parseCommandLine(args)) return false;
+    if (!verifyRequiredArgs(args)) return false;
+    return true;
+  }
+
+  private boolean verifyRequiredArgs(String[] args) {
+    if (sourceZkCluster == null
+        && (sourceBigtableProjectId == null || sourceBigtableInstance == null)) {
+      printUsage(
+          "--sourcezkcluster or --sourcebigtableproject and --sourcebigtableinstance required.",
+          args);
+      return false;
+    }
+
+    if (sourceHashDir == null) {
+      printUsage("--sourcehashdir is required.", args);
+      return false;
+    }
+
+    if (sourceTableName == null) {
+      printUsage("--sourcetablename is required.", args);
+      return false;
+    }
+
+    if (targetTableName == null) {
+      printUsage("--targettablename is required.", args);
+      return false;
+    }
+
+    if (syncOutputDir == null) {
+      printUsage("--outputpath is required.", args);
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean parseCommandLine(String[] args) {
+    try {
+      for (int i = 0; i < args.length; i++) {
+        String cmd = args[i];
+        if (cmd.equals("-h") || cmd.startsWith("--h")) {
+          printUsage(null, args);
+          return false;
+        }
+
+        // required
+        String sourceZkClusterKey = "--sourcezkcluster=";
+        if (cmd.startsWith(sourceZkClusterKey)) {
+          sourceZkCluster = cmd.substring(sourceZkClusterKey.length());
+          continue;
+        }
+
+        String sourceBigtableProjectIdKey = "--sourcebigtableproject=";
+        if (cmd.startsWith(sourceBigtableProjectIdKey)) {
+          sourceBigtableProjectId = cmd.substring(sourceBigtableProjectIdKey.length());
+          continue;
+        }
+
+        String sourceBigtableInstanceKey = "--sourcebigtableinstance=";
+        if (cmd.startsWith(sourceBigtableInstanceKey)) {
+          sourceBigtableInstance = cmd.substring(sourceBigtableInstanceKey.length());
+          continue;
+        }
+
+        String sourceBigtableAppProfileKey = "--sourcebigtableprofile=";
+        if (cmd.startsWith(sourceBigtableAppProfileKey)) {
+          sourceBigtableAppProfile = cmd.substring(sourceBigtableAppProfileKey.length());
+          continue;
+        }
+
+        String targetZkClusterKey = "--targetzkcluster=";
```

Comment:
should usage() share the definitions with this method? They are both relying on common parameters --targetzkcluster